pi/interfaces.txt

```diff
@@ -9,13 +11,13 @@ iface lo inet loopback
 auto FLIC_HUB_INTERFACE
 iface FLIC_HUB_INTERFACE inet manual
 
-auto wlan0
-iface wlan0 inet dhcp
-wpa-conf /etc/wpa_supplicant/wpa_supplicant.conf
-  post-up /sbin/ip link set wlan0 promisc on
-  post-up /usr/sbin/parprouted FLIC_HUB_INTERFACE wlan0
+auto NETWORK_INTERFACE
+iface NETWORK_INTERFACE inet dhcp
+WPA_CONF
+  post-up /sbin/ip link set NETWORK_INTERFACE promisc on
```

Comment:
Does this indentation represent anything functional based on `WPA_CONF`?

Reply:
so `/etc/network/interfaces` has its own syntax and I'm not familiar enough with it to know the answer to this. what I do know is that all the example code that I've seen has this indentation scheme

Reply:
OK. I was just curious whether the indented lines should still appear when `WPA_CONF` is the empty string.

Reply:
my understanding is that the `post-up` and `post-down` hooks are part of the `iface` stanza along with the `wpa-conf` (if present). welcome to weird linux networking syntax.

from the interfaces manpage (run `man interfaces` on a RPi):

```
The following "command" options are available for every family and method.  Each of these options can be given multiple times in a single stanza, in which case the commands are executed in the order in which they appear in the stanza.

pre-up ...
```